resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -175,69 +175,61 @@ private[spark] class KubernetesClusterSchedulerBackend(
-        .withLabels(selectors)
-        .withOwnerReferences()
-        .addNewOwnerReference()
-          .withController(true)
-          .withApiVersion(driverPod.getApiVersion)
-          .withKind(driverPod.getKind)
-          .withName(driverPod.getMetadata.getName)
-          .withUid(driverPod.getMetadata.getUid)
-        .endOwnerReference()
-      .endMetadata()
-      .withNewSpec()
-        .addNewContainer()
-          .withName(s"exec-${applicationId()}-container")
-          .withImage(executorDockerImage)
-          .withImagePullPolicy("IfNotPresent")
-          .withNewResources()
-            .addToRequests("memory", executorMemoryQuantity)
-            .addToLimits("memory", executorMemoryLimitQuantity)
-            .addToRequests("cpu", executorCpuQuantity)
-            .addToLimits("cpu", executorCpuQuantity)
-            .endResources()
-          .withEnv(requiredEnv.asJava)
-          .withPorts(requiredPorts.asJava)
-          .endContainer()
-        .endSpec()
-      .done())
+    val requiredEnv = Seq(
+      (ENV_EXECUTOR_PORT, executorPort.toString),
+      (ENV_DRIVER_URL, driverUrl),
+      (ENV_EXECUTOR_CORES, executorCores),
+      (ENV_EXECUTOR_MEMORY, executorMemory),
+      (ENV_APPLICATION_ID, applicationId()),
+      (ENV_EXECUTOR_ID, executorId)
+    ).map(env => new EnvVarBuilder()
+      .withName(env._1)
+      .withValue(env._2)
+      .build())
+    val requiredPorts = Seq(
+      (EXECUTOR_PORT_NAME, executorPort),
+      (BLOCK_MANAGER_PORT_NAME, blockmanagerPort))
+      .map(port => {
+        new ContainerPortBuilder()
+          .withName(port._1)
+          .withContainerPort(port._2)
+          .build()
+      })
+    try {
+      (executorKubernetesId, kubernetesClient.pods().createNew()
+        .withNewMetadata()
+          .withName(name)
+          .withLabels(selectors)
+          .withOwnerReferences()
+          .addNewOwnerReference()
+            .withController(true)
+            .withApiVersion(driverPod.getApiVersion)
+            .withKind(driverPod.getKind)
+            .withName(driverPod.getMetadata.getName)
+            .withUid(driverPod.getMetadata.getUid)
+          .endOwnerReference()
+        .endMetadata()
+        .withNewSpec()
+          .addNewContainer()
+            .withName(s"executor")
+            .withImage(executorDockerImage)
+            .withImagePullPolicy("IfNotPresent")
+            .withNewResources()
+              .addToRequests("memory", executorMemoryQuantity)
+              .addToLimits("memory", executorMemoryLimitQuantity)
+              .addToRequests("cpu", executorCpuQuantity)
+              .addToLimits("cpu", executorCpuQuantity)
+              .endResources()
+            .withEnv(requiredEnv.asJava)
+            .withPorts(requiredPorts.asJava)
+            .endContainer()
+          .endSpec()
+        .done())
+    } catch {
+      case throwable: Throwable =>
+        logError("Failed to allocate executor pod.", throwable)
```

Comment:
In testing and running into problems I found that the exception here doesn't appear in the logs. There might be something in the calling code worth fixing instead of logging at this layer.